modules/layers/src/layers/editable-h3-cluster-layer.ts

```diff
@@ -0,0 +1,180 @@
+        this.getSubLayerProps({
+          id: 'editable-geojson',
+
+          mode: this.props.mode,
+          data: EMPTY_FEATURE_COLLECTION,
+          selectedFeatureIndexes: [],
+
+          onEdit: (editAction) => {
+            const { editType, editContext } = editAction;
+
+            switch (editType) {
+              case 'updateTentativeFeature':
+                // tentative feature updates, updated on every pointer move
+                if (editContext.feature.geometry.type === 'Polygon') {
+                  const coords = editContext.feature.geometry.coordinates;
+                  const hexIDs = this.getDerivedHexagonIDs(coords);
+
+                  this.setState({ tentativeHexagonIDs: hexIDs });
+                } else if (editContext.feature.geometry.type === 'Point') {
+                  const coords = editContext.feature.geometry.coordinates;
+                  const hexID = this.getDerivedHexagonID(coords);
+
+                  this.setState({ tentativeHexagonIDs: [hexID] });
+                }
+                break;
+              case 'addFeature':
+                // once the shape is finished drawing, calculate final hexagons
+
+                // get selected h3 cluster to act upon
+                // if there is more than 1 or 0 selected,
+                // throw an error, since it is ambiguous how to proceed
+                // TODO: Add support for getNewH3Cluster prop to create a new cluster if none exist
+                if (
+                  this.props.selectedIndexes.length > 1 ||
+                  this.props.selectedIndexes.length === 0
+                ) {
+                  // eslint-disable-next-line no-console,no-undef
+                  console.warn('booleanOperation only supported for single cluster selection');
+                } else {
+                  let finalHexagonIDs;
+                  const committedHexagonIDs = new Set(this.getSelectedHexIDs());
+                  const tentativeHexagonIDs = new Set(this.state.tentativeHexagonIDs);
+
+                  const { modeConfig } = this.props;
+                  switch (modeConfig && modeConfig.booleanOperation) {
+                    case 'union':
+                    default:
+                      finalHexagonIDs = [
+                        ...new Set([...committedHexagonIDs, ...tentativeHexagonIDs]),
+                      ];
+                      break;
+                    case 'intersection':
+                      finalHexagonIDs = [...committedHexagonIDs].filter((hexID: string) =>
+                        tentativeHexagonIDs.has(hexID)
+                      );
+                      break;
+                    case 'difference':
+                      finalHexagonIDs = [...committedHexagonIDs].filter(
+                        (hexID: string) => !tentativeHexagonIDs.has(hexID)
+                      );
+                      break;
+                  }
+                  this.props.onEdit({ updatedData: finalHexagonIDs });
```

Comment:
To match `EditableGeoJsonLayer`, this should include all the clusters, not just the affected cluster. This way, users can simplify their `onEdit` callback to this:
```
const [data, setData] = React.useState([...]);

onEdit({updatedData}) {
  if (updatedData !== data) {
    setData(updatedData);
  }
}
```

So, basically take the logic you currently have in the example app, and put it here.